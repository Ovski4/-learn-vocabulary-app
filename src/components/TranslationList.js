import React from 'react';
import TranslationEditionRow from './edition/TranslationEditionRow';
import TranslationRevisionRow from './revision/TranslationRevisionRow';
import { Text, TextInput, View, FlatList } from 'react-native';

class TranslationList extends React.PureComponent {

    render() {
        const RowComponents = {
            'edition': TranslationEditionRow,
            'revision': TranslationRevisionRow
        }

        const getListItem = (translation) => {
            return React.createElement(
                    RowComponents[this.props.mode],
                    {
                        key: translation.createdAt,
                        translation: translation
                    }
                )
        };

        return (
            <View>
                <View style={{
                    alignItems: 'center',
                    margin: 5

                }}>
                    <Text style={{fontWeight: 'bold'}}>Translations</Text>
                </View>
                {/* <View style={{
                        backgroundColor: '#eeeeee',
                        borderRadius: 2,
                        paddingLeft: 5,
                        marginLeft: 5,
                        marginRight: 5,
                        marginBottom: 5
                    }}>
                        <TextInput
                            underlineColorAndroid="transparent"></TextInput>
                </View> */}
                <View style={{height: 350}}>
                    <FlatList
                        data={this.props.translations}
                        extraData={this.props.mode}
                        renderItem={({item}) => getListItem(item)}
                        keyExtractor={(item, index) => index.toString()}
                    />
                </View>
            </View>
        );
    }
}

export default TranslationList;
