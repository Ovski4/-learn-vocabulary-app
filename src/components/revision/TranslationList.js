import React from 'react';
import TranslationRow from './TranslationRow';
import translationsService from '../../services/translations'
import { StyleSheet, Text, View, FlatList } from 'react-native';
import Header from '../ui/Header';
import { connect } from 'react-redux';

const mapStateToProps = (state) => {
    return {
        ui: state.ui,
        translations: [...state.translations].reverse()
    }
};

const styles = StyleSheet.create({
    emptyListView: {
        flexDirection: 'row',
        borderTopWidth: 0.4,
        borderColor: '#bbb',
        justifyContent: 'center'
    },
    emptyListText: {
        fontStyle: 'italic',
        lineHeight: 50
    }
});

class TranslationList extends React.PureComponent {

    render() {
        const getListItem = (translation) => {
            return <TranslationRow
                key={translation.id}
                id={translation.id}
                hidden={translation.hidden}
            />;
        };

        const listEmptyComponent =
            <View style={styles.emptyListView}>
                <Text style={styles.emptyListText}>
                    Add a translation to start learning!
                </Text>
            </View>
        ;

        const tagFilterId = this.props.ui.translationsFilteredBy;
        const translationList = tagFilterId !== null ?
            translationsService.filterByTagId(this.props.translations, tagFilterId) :
            this.props.translations
        ;

        return(
            <View style={{flex: 1}}>
                <Header>Translations</Header>
                <View style={{
                    flex: 1,
                    borderColor: '#bbb'
                }}>
                    <FlatList
                        data={translationList}
                        index={({item}) => item.index}
                        renderItem={({item}) => getListItem(item)}
                        keyExtractor={(item, index) => item.id}
                        ListEmptyComponent={listEmptyComponent}
                    />
                </View>
            </View>
        );
    }
}

export default connect(mapStateToProps)(TranslationList);
