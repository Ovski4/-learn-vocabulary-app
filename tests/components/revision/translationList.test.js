import reducer from '../../../src/reducers/index';
import { createStore } from 'redux';
import React from 'react';
import TranslationList from '../../../src/components/revision/translationList/connect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

describe('TranslationList component', () => {

    it('Should render the component', () => {

        const store = createStore(reducer, {
            translations: [],
            config: {
                locale: 'en'
            }
        });

        const rendered = renderer.create(<Provider store={store}><TranslationList /></Provider>).toJSON();
        expect(rendered).toMatchSnapshot();
    });

});
