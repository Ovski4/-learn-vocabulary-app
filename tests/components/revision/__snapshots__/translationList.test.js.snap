// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationList component Should render the component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Translations
    </Text>
  </View>
  <View
    style={
      Object {
        "borderColor": "#bbb",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <Component
          style={
            Object {
              "borderColor": "#bbb",
              "borderTopWidth": 0.4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontStyle": "italic",
                "lineHeight": 50,
              }
            }
          >
            Add a translation to start learning!
          </Text>
        </Component>
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      index={[Function]}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderColor": "#bbb",
                "borderTopWidth": 0.4,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontStyle": "italic",
                  "lineHeight": 50,
                }
              }
            >
              Add a translation to start learning!
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
